server/src/routes/donatedItemRoutes.ts

```diff
@@ -8,7 +8,28 @@ const router = Router();
 // POST /donatedItem - Create a new DonatedItem
 router.post('/', donatedItemValidator, async (req: Request, res: Response) => {
     try {
+        
         const { dateDonated, ...rest } = req.body;
+        
+        // Validate donorId
+        const donorExists = await prisma.donor.findUnique({
+            where: { id: req.body.donorId }
+        });
+
+        
+        if (!donorExists) {
+            return res.status(400).json({ error: "Donor ID is not valid or does not exist." });
+        }
+
+        // Validate programId
+        const programExists = await prisma.program.findUnique({
+            where: { id: req.body.programId }
+        });
+
+        
+        if (!programExists) {
+            return res.status(400).json({ error: "Program ID is not valid or does not exist." });
+        }
 
```

Comment:
I have seen the donor and program validation in both POST and PUT requests. Can we consolidate it so that there will be no duplicate code?